### PR TITLE
Speed up the initial menu filling

### PR DESCRIPTION
**Pull request by banyaszg**
Speeding up the init of the menus.
- [x] Menu interface
- [x] FileChoice::openMenu()
- [x] Choice::fillMenu()

In EdgeTX (these will be in an other PR):
- [x] ModelSetupPage::build()
- [x] ModelsPageBody::editLabels()
- [x] ModelCurvesPage::build()
- [x] ModelMixesPage::newMix()

### CODE REVIEW

File: src/menu.cpp [hunk 1]
```diff
@@ -312,19 +317,27 @@ void Menu::setTitle(std::string text)
...
+}
+
 void Menu::addLine(const std::string &text, std::function<void()> onPress,
-                   std::function<bool()> isChecked)
+                   std::function<bool()> isChecked, 
+                   bool updatePos)
 {
   content->body.addLine(text, std::move(onPress), std::move(isChecked));
-  updatePosition();
+  if(updatePos) updatePosition();
```
Comment (gagarinlg) [inline src/menu.cpp:331]: Can you please put those into a new line? this kind of code formating is OK if you have multiple one line ifs together, but written like that it is dangerous (ask me how I know...)

File: src/menu.h [hunk 2]
```diff
@@ -103,7 +104,7 @@ class MenuBody: public TableField
                  std::function<bool()> isChecked);
 
     void removeLines();
-
+
```
Comment (gagarinlg) [inline src/menu.h:107]: why? ;)

**Review by gagarinlg — CHANGES_REQUESTED** [review]

**Comment by banyaszg** [comment]
@gagarinlg
Design question: The Menu::addLine() function calls the Menu::updatePosition() every time which is unnecessary if we do it in a cycle. It would be enough to do it at the end. I see two solutions:
1. Adding a parameter to the Menu:addLine() to disable the update in these cases.
2. Or creating a flag with a set function to disable the update before the mass add, and reenable it at the end.

File: src/menu.cpp (hunk 1 above)
Comment (banyaszg) [reply src/menu.cpp:331]: Ok, I'll do :)

Comment (gagarinlg) [reply src/menu.cpp]: I could have lived without the brackets. but they donÄt bother me much. So for me this is OK like you did it now.

**Comment by gagarinlg** [comment]
> @gagarinlg Design question: The Menu::addLine() function calls the Menu::updatePosition() every time which is unnecessary if we do it in a cycle. It would be enough to do it at the end. I see two solutions:
> 
>     1. Adding a parameter to the Menu:addLine() to disable the update in these cases.
> 
>     2. Or creating a flag with a set function to disable the update before the mass add, and reenable it at the end.

I have no real preference, it depends on the use case. So I do not see any value in changing the implementation you already did.

**Comment by banyaszg** [comment]
It seems to me some other parts has mass addLine() calls too:
- Choice::fillMenu()
- ModelMixesPage::newMix()

Should we modify them too? I think Choice is base class of several other classes.

**Comment by gagarinlg** [comment]
> It seems to me some other parts has mass addLine() calls too:
> 
>     * Choice::fillMenu()
> 
>     * ModelMixesPage::newMix()
> 
> 
> Should we modify them too? I think Choice is base class of several other classes.

Absolutely, go ahead.

**Comment by banyaszg** [comment]
Also I'm not sure the setLineCount() is the right choice. Perhaps I should implement instead an addLineCount() and call that before any mass addLine() calls.

**Comment by gagarinlg** [comment]
Maybe adding a new function, that gets a count and a functor to get the names of the new entries?
The we have two functions, addLine for one line and addLines for many new lines

**Comment by banyaszg** [comment]
Sounds a nice, clean solution. I'll check how could I implement the functionalities with such an interface.

**Comment by banyaszg** [comment]
> Maybe adding a new function, that gets a count and a functor to get the names of the new entries? The we have two functions, addLine for one line and addLines for many new lines

I tried to redesign the code using addLines() functions. Actually there are several attributes for a line entry, so I had to use a function with several output parameters to set everything.
Also I checked the referencing parts so I tried to include every cases in the interface.

I really hate the code redundancy in the two addLines() functions, but I don't have any good idea to fix that. I could do it with define, but that also doesn't looks good.

File: src/menu.cpp (hunk 1 above)
Comment (pfeerick) [reply src/menu.cpp]: Braces are good... it also tells the auto-formatter to not put short lines on one line ;)

**Comment by banyaszg** [comment]
Now it looks better.

**Comment by banyaszg** [comment]
> Maybe adding a new function, that gets a count and a functor to get the names of the new entries? The we have two functions, addLine for one line and addLines for many new lines

@gagarinlg I tried the "addLines" implementation but it was too complicated to cover all the use cases. So at last I implemented this simple interface. It does a delayed lvgl update.And it requires only a minor modifcation in the current codes.

**Pull request merged**